src/Survey/ItechArt.Survey.Foundation/Authentication/Abstractions/IAuthenticateService.cs

```diff
@@ -8,5 +8,7 @@ public interface IAuthenticateService
 {
     Task<OperationResult<User, UserRegistrationErrors>> RegisterAsync(User user, string password);
 
+    Task<OperationResult<string, UserAuthenticationErrors>> AuthenticateAsync(string userName, string password);
```

Comment:
Adjust return type solutions:
1) Revisit OperationResult<TError> to support errors only (OperationResult<TResult, TError> should still be present) - let's go with this one
2) UserAuthenticationErrors?
3) UserAuthenticationStatus = Successful + UserAuthenticationErrors
4) bool